streatham_go/app/views/leaderboard.py

```diff
@@ -1,7 +1,42 @@
 from django.shortcuts import render
 from django.contrib.auth.decorators import login_required
+from django.contrib.auth.models import User
+from ..models import Leaderboard
 
 
 @login_required
 def leaderboard(request):
-    return render(request, 'app/leaderboard.html')
+    leaderboard_data = Leaderboard.objects.order_by('-level', 'quiz_count')
+    users = User.objects.all()
+    user_data = []
+    for user in users:
+        user_leaderboard_data = leaderboard_data.filter(user=user).first()
+        user_data.append({
+            'username':
+            user.username,
+            'level':
+            user_leaderboard_data.level if user_leaderboard_data else 1,
+            'quiz_count':
+            user_leaderboard_data.quiz_count if user_leaderboard_data else 0
+        })
+
+    user_data = sorted(user_data, key=lambda x: x['level'], reverse=True)
+
+    # Find the current user's leaderboard data and rank
+    current_user_data = leaderboard_data.filter(user=request.user).first()
+    current_user_rank = user_data.index(
```

Comment:
Please move this into a function in the Leaderboard class

Reply:
Moved to separate function